src/fhir/document.ts

```diff
@@ -1,33 +1,74 @@
 import { searchCommonRecords } from "./search-helpers";
-import { useQueryWithPatient } from "..";
+import { PatientModel, useQueryWithPatient } from "..";
 import { applyDocumentFilters } from "@/components/content/document/helpers/filters";
+import { CTWRequestContext } from "@/components/core/providers/ctw-context";
+import { createGraphqlClient } from "@/services/fqs/client";
+import { DocumentReferenceGraphqlResponse, documentsQuery } from "@/services/fqs/queries/documents";
 import { orderBy } from "@/utils/nodash";
 import { QUERY_KEY_PATIENT_DOCUMENTS } from "@/utils/query-keys";
 import { Telemetry, withTimerMetric } from "@/utils/telemetry";
 
-export function usePatientDocument() {
+export function usePatientDocument(enableFQS: boolean) {
   return useQueryWithPatient(
     QUERY_KEY_PATIENT_DOCUMENTS,
     [],
-    withTimerMetric(async (requestContext, patient) => {
-      try {
-        const { resources } = await searchCommonRecords("DocumentReference", requestContext, {
-          patientUPID: patient.UPID,
-        });
-        const results = orderBy(
-          applyDocumentFilters(resources),
-          [(document) => document.resource.content[0].attachment.creation || ""],
-
-          ["desc"]
-        );
-        if (results.length === 0) {
-          Telemetry.countMetric("req.count.documents.none");
-        }
-        Telemetry.histogramMetric("req.count.documents", results.length);
-        return results;
-      } catch (e) {
-        throw new Error(`Failed fetching document information for patient: ${e}`);
-      }
-    }, "req.timing.documents")
+    enableFQS
+      ? withTimerMetric(
+          async (requestContext, patient) => getDocumentFromFQS(requestContext, patient),
+          "req.timing.documents",
+          ["FQS"]
```

Comment:
don't know if capitalization matters, but everywhere else this is `fqs`
```suggestion
          ["fqs"]
```